core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -491,7 +491,11 @@ struct CudaParallelLaunchKernelInvoker<
             cuda_instance->m_deviceProp, block_size, shmem, desired_occupancy);
       }
 
-      auto* driver_ptr = Impl::allocate_driver_storage_for_kernel(driver);
+      // This cannot be merged.
+      auto* driver_ptr = Impl::allocate_driver_storage_for_kernel(
+          CudaSpace::impl_create(cuda_instance->m_cudaDev,
+                                 cuda_instance->m_stream),
+          driver);
```

Comment:
Why not?

Reply:
Well, I wanted to discuss why at this point we're dealing with a `CudaInternal` and not a `Cuda` or a `CudaSpace`, but if the code looks OK for you then I'm OK with it.